src/MANYBODY/pair_eam.cpp

```diff
@@ -237,7 +239,9 @@ void PairEAM::compute(int eflag, int vflag)
     fp[i] = (coeff[0]*p + coeff[1])*p + coeff[2];
     if (eflag) {
       phi = ((coeff[3]*p + coeff[4])*p + coeff[5])*p + coeff[6];
-      if (rho[i] > rhomax) phi += fp[i] * (rho[i]-rhomax);
+      if (rho[i] > rhomax) {
```

Comment:
did you mean to set or increment beyond_rhomax here ?

Reply:
Does this need to be ported to Kokkos?

Reply:
> did you mean to set or increment beyond_rhomax here ?

Yes. No idea how it vanished.

Reply:
> Does this need to be ported to Kokkos?

Yes, it could be. I have not looked too deeply at implementing it myself (it was too late for KOKKOS).
It is only an additional diagnostic for some rare and obscure use cases, so I am not sure whether it would be needed in KOKKOS. I would expect that people paying attention to the warning would be running without KOKKOS first. I mostly put the code there so that the others that were discussing it in the issue page would see what it could look like. I already set it up so the check is only on energy output and thus the collective communication is not in every step by default.
I would be happy to revert it and only keep the warning in the manual.